src/lib/journey/callbacks/select-idp/select-idp.svelte

```diff
@@ -0,0 +1,71 @@
+<script lang="ts">
+  import type { SelectIdPCallback } from '@forgerock/javascript-sdk';
+  import AppleIcon from '../../../components/icons/apple-icon.svelte';
+  import Button from '$components/primitives/button/button.svelte';
+  import Grid from '$components/primitives/grid/grid.svelte';
+  import { interpolate, textToKey } from '$lib/_utilities/i18n.utilities';
+
+  import type {
+    CallbackMetadata,
+    SelfSubmitFunction,
+    StepMetadata,
+  } from '$journey/journey.interfaces';
+  import type { Style } from '$lib/style.store';
+  import type { Maybe } from '$lib/interfaces';
+
+  export let callback: SelectIdPCallback;
+  export let callbackMetadata: CallbackMetadata;
+  export let selfSubmitFunction: Maybe<SelfSubmitFunction> = null;
+  export let stepMetadata: StepMetadata;
+  export let style: Style = {};
+
+  let inputName: string;
+  let label: string;
+  let idps: { value: string; text: string }[];
+  let buttonStyle: 'outline' | 'primary' | 'secondary' | undefined;
+
+  /**
+   * @function setButtonValue - Sets the value on the callback on button click
+   * @param {number} index
+   */
+  function setBtnValue(value: string) {
+    callback.setProvider(value);
+    callbackMetadata.isReadyForSubmission = true;
+    selfSubmitFunction && selfSubmitFunction();
+  }
+
+  $: {
+    const localAuthentication = callback
+      .getProviders()
+      .filter((provider) => provider.provider === 'localAuthentication');
+
+    const socialProviders = callback
+      .getProviders()
+      .filter((provider) => provider.provider !== 'localAuthentication');
+
+    if (localAuthentication.length > 0) {
+      // Assume that clicking "next" will indicate the user wants to use local authentication
+      callback.setProvider('localAuthentication');
+    }
+
+    idps = socialProviders.map((option, index) => ({
+      value: option.provider,
+      text: option.uiConfig.buttonDisplayName,
+    }));
+  }
+</script>
+
+<Grid num={idps.length}>
+  {#each idps as idp}
+    <Button type="button" width="auto" onClick={() => setBtnValue(idp.value)}
+      ><AppleIcon
+        classes="tw_inline-block tw_fill-current tw_text-secondary-dark dark:tw_text-secondary-light"
```

Comment:
In the demo to the team the look and feel of the Sign In With Apple button was raised, given that Apple have UX guidelines for this button: https://developer.apple.com/design/human-interface-guidelines/technologies/sign-in-with-apple/

There was also an unresolved discussion about whether we should build the callback to meet these requirements, or consume the uiconfig from AM. @ryanbas21 and I decided that that for this story we will treat the styling of the button as a non-blocking nice-to-have item and address the UX for all of the social buttons in the following story where other IDPs are added.

Reply:
UX of the social buttons has been added to the following social story where we will add other providers.